crypto/keys/src/ed25519.rs

```diff
@@ -397,6 +401,24 @@ impl PartialEq for Ed25519Signature {
     }
 }
 
+impl PartialOrd for Ed25519Signature {
```

Comment:
Might `derive_core_cmp_from_as_ref!` be useful here?

Reply:
I forgot we had that, I'll take a look. Thanks.
How did you find out about its existence?

Reply:
One potential reason to not use the macro is that it implements `PartialEq` as well:
```
        impl PartialEq for $mytype {
            fn eq(&self, other: &Self) -> bool {
                <Self as AsRef<$asref>>::as_ref(self).eq(<Self as AsRef<$asref>>::as_ref(other))
            }
        }
```

however, we are able to provide a more efficient implementation for that since the underlying object implements that a well:
```
impl PartialEq for Ed25519Signature {
    fn eq(&self, other: &Self) -> bool {
        self.0.eq(&other.0)
    }
}
```

Since comparing signatures is a pretty common operation, I am no longer sure using the macro is the right move.

Any thoughts?

Reply:
Ah, fair point. Having the specialized impl SGTM then.

> I forgot we had that, I'll take a look. Thanks. How did you find out about its existence?

I basically stumbled upon it while studying our macro usage.